expression/aggregation/base_func.go

```diff
@@ -305,6 +305,24 @@ var noNeedCastAggFuncs = map[string]struct{}{
 	ast.AggFuncJsonObjectAgg:       {},
 }
 
+// We need to wrap cast as decimal upon these functions,
+// since we need to keep the fraction part of their decimal results
+var needCastAsDecimalAggFuncs = map[string]struct{}{
```

Comment:
Will other agg func have the same cast needs?

Reply:
I think `group_conat` is the only one.